src/spellCheckerProvider.ts

```diff
@@ -112,6 +112,21 @@ class SpellCheckerProvider {
     return spellChecker.suggest(text);
   }
 
+  /**
+   * Add a word to the current dictionary.
+   * Runtime only -- added words do not persist between sessions!
+   * @param {string} text: word to be added
+   * @returns {Promise<void>} Indication to load completes.
+   */
+  public async addWord(text: string): Promise<void> {
```

Comment:
thinking bit, I think this should be

```
addWord(dictionaryKey: string, text: string)
```

to specify which dictionary we'll add this word into instead of implicitly rely on a currently selected spellchecker.

Reply:
Hmm.. I don't think there's support for this in [hunspell-asm](https://github.com/kwonoj/hunspell-asm)?   Specifying a language other than the current one would involve switching dictionaries, which could get messy..

Reply:
`electron-hunspell` creates each hunspell instance per dictionary specified. So `hunspell-asm` does not aware around what kind of dict.

Problem of current PR is it implicitly rely on `currently selected dict` to add word. If you have multiple dictionary loaded, you may want to specify which dict you'd like to add word / or all dict.

Reply:
To add word for all dictionary, there's no need to `switch`: just need to lookup spellcheckertable would be sufficient as we don't expect any spellchecker behavior change on provider side.

Reply:
Ah gotcha, thanks, I see that now!